type-metadata/src/impls.rs

```diff
@@ -105,7 +105,7 @@ where
 	T: HasTypeId,
 {
 	fn type_id() -> TypeId {
-		<[T]>::type_id()
+		TypeIdCustom::new("Vec", Namespace::prelude(), tuple_type_id![T]).into()
```

Comment:
What difference in type id and def?

Reply:
If I did not miss something, JS side has no concept of "vec vs slice". Then this will be a special case for JS, "while handling custom type ids, a custom `Vec` needed to be treated as a JS array."

If we decide not to expose other rust specific stuff like Box or Ref, we should not expose this rust specifics either, namely vec vs slice. Though if their difference matters in any cases we could think of, they should definitely have different type ids and defs. The same logic applies to `&str` vs `String`.

Reply:
I spoke to our JS side about this.
They say it would be easy to identify the `Vec` <=> `slice` relationship upon decoding so it shouldn't be that bad.
Still I think it would be better if we do not introduce any special cases into our encoding.
However, I also understand why you would want that - to simplify for the common cases.
I am not settled upon any decision we take so let's defer this decision at a later point where it makes more sense.